### DISCUSSION TITLE
SML script with filesystem brings wrong values

### DISCUSSION THREAD

**Discussion by killah78** [arendst/Tasmota#21646] [Support request]
Hi, I need some help in understanding. 
I want to create a tasmota (currently testing on nodemcuV3, but later I want to bring it on ESP-12F) for smart meter reading. Mainly for 9 mbus devices on one mbus.
I downloaded the tasmota-scripting.bin from Jason2866. 
This is working fine. Regarding the debugoutput I created the decoders. And the output is fine. 
But as I have 9 devices, I can not bring everything to the script. I need more space. 
So I compiled an own tasmota on gitpod, adding 
USE_SCRIPT, USE_SML_M and USE_UFILESYS and UFSYS_SIZE 6500. I also added board_build.ldscript = eagle.flash.4m2m.ld
It compiled fine and I can see it is with filesystem and I have more space in the script. 

Now comes the issue:
I copied exactly the same script, but get different results. The first rows are ok, then the decoder bring wrong values. 

See below first working one, second the wrong results. 

I have no idea what is wrong. 
Appreciate any hint on what could be the issue.

`working:

script:

>D 
>B 
=>sensor53 r
tper=10
>M 1 
+1,3,rE1,0,2400,MBUS,1,50,10400b4b16,107b0b8616,68030368530bB10f16
1,685d5d68080b72bcd8@1,11-Zähler-Nr.,,11_zähler,0
1,685d5d68080b72x14bcd8@100,11-total active energy,kWh,11_total_active_energy,2
1,685d5d68080b72x20bcd8@100,11-import active energy,kWh,11_import_active_energy,2
1,685d5d68080b72x26bcd8@100,11-export active energy,kWh,11_export_active_energy,2
1,685d5d68080b72x32bcd8@100,11-r.total active e,kWh,11_res_total_active_energy,2
1,685d5d68080b72x38bcd8@100,11-r.import active e,kWh,11_res_import_active_energy,2
1,685d5d68080b72x44bcd8@100,11-r.export active e,kWh,11_res_export_active_energy,2
1,685d5d68080b72x50bcd8@1,11-total reactive e,kWh,11_total_reactive_energy,1
1,685d5d68080b72x56bcd8@1,11-import reactive e,kWh,11_import_reactive_energy,1
1,685d5d68080b72x62bcd8@1,11-export reactive e,kWh,11_export_reactive_energy,1
1,685d5d68080b72x68bcd8@1,11-r. total reactive e,kWh,11_res_total_reactive_energy,1
1,685d5d68080b72x74bcd8@1,11-r. import reactive e,kWh,11_res_import_reactive_energy,1
1,685d5d68080b72x80bcd8@1,11-r. export reactive e,kWh,11_res_export_reactive_energy,1
1,68909068080b72x15bcd6@100,11-L1 voltage,V,11_L1_voltage,2
1,68909068080b72x21bcd6@100,11-L2 voltage,V,11_L2_voltage,2
1,68909068080b72x27bcd6@100,11-L3 voltage,V,11_L3_voltage,2
1,68909068080b72x33bcd6@100,11-L1L2 voltage,V,11_L1L2_voltage,2
1,68909068080b72x39bcd6@100,11-L2L3 voltage,V,11_L2L3_voltage,2
1,68909068080b72x45bcd6@100,11-L3L1 voltage,V,11_L3L1_voltage,2
1,68909068080b72x51bcd6@1000,11-L1 current,A,11_L1_current,2
1,68909068080b72x57bcd6@1000,11-L2 current,A,11_L2_current,2
1,68909068080b72x63cd6@1000,11-L3 current,A,11_L3_current,2
1,68909068080b72x69bcd6@1000,11-N current,A,11_N_current,2
1,68909068080b72x74bcd6@10,11-total active power,W,11_total_active_power,2
1,68909068080b72x79bcd6@10,11-L1 active power,W,11_L1_active_power,2
1,68909068080b72x84bcd6@10,11-L2 active power,W,11_L2_active_power,2
1,68909068080b72x89bcd6@10,11-L3 active power,W,11_L3_active_power,2
1,68909068080b72x119bcd4@100,11-total power factor,,11_total_power_factor,2
1,68909068080b72x139bcd4@100,11-frequency,,11_frequency,2

dump:
15:38:40.111 : 68 5d 5d 68 08 0b 72 65 79 24 01 24 40 01 02 55 00 00 00 0c 04 86 85 28 02 0c 04 29 80 28 02 0c 
15:38:40.256 : 04 57 05 00 00 0c 04 86 85 28 02 0c 04 29 80 28 02 0c 04 57 05 00 00 0c fd 3a 36 88 77 00 0c fd 
15:38:40.403 : 3a 42 48 13 00 0c fd 3a 93 39 64 00 0c fd 3a 36 88 77 00 0c fd 3a 42 48 13 00 0c fd 3a 93 39 64 
15:38:40.457 : 00 02 16 
15:38:45.783 : 68 90 90 68 08 0b 72 65 79 24 01 24 40 01 02 55 00 00 00 0b fd 47 77 36 02 0b fd 47 43 37 02 0b 
15:38:45.929 : fd 47 82 36 02 0b fd 47 68 10 04 0b fd 47 72 10 04 0b fd 47 15 10 04 0b fd 59 75 20 00 0b fd 59 
15:38:46.073 : 34 08 00 0b fd 59 20 10 00 0b fd 59 47 11 00 0b 2a 06 72 00 0b 2a 68 36 00 0b 2a 88 18 00 0b 2a 
15:38:46.220 : 70 16 00 0b fd 3a 14 13 f0 0b fd 3a 19 05 f0 0b fd 3a 85 02 00 0b fd 3a 83 10 f0 0a fd 3a 83 09 
15:38:46.361 : 0a fd 3a 90 09 0a fd 3a 88 09 0a fd 3a 38 08 0a fd 3a 99 49 6f 16 

output:
MBUS 11-Zähler-Nr.	1247965
MBUS 11-total active energy	22885.84 kWh
MBUS 11-import active energy	22880.27 kWh
MBUS 11-export active energy	5.57 kWh
MBUS 11-r.total active e	22886 kWh
MBUS 11-r.import active e	22880 kWh
MBUS 11-r.export active e	6 kWh
MBUS 11-total reactive e	77883540.0 kWh
MBUS 11-import reactive e	48424163 kWh
MBUS 11-export reactive e	93416312 kWh
MBUS 11-r. total reactive e	41631200 kWh
MBUS 11-r. import reactive e	163120077 kWh
MBUS 11-r. export reactive e	12001348 kWh
MBUS 11-L1 voltage	236.90 V
MBUS 11-L2 voltage	237.06 V
MBUS 11-L3 voltage	236.04 V
MBUS 11-L1L2 voltage	410.46 V
MBUS 11-L2L3 voltage	409.71 V
MBUS 11-L3L1 voltage	409.58 V
MBUS 11-L1 current	2.09 A
MBUS 11-L2 current	0.84 A
MBUS 11-L3 current	0.00 A
MBUS 11-N current	1.16 A
MBUS 11-total active power	728.20 W
MBUS 11-L1 active power	372.00 W
MBUS 11-L2 active power	190.20 W
MBUS 11-L3 active power	168.10 W
MBUS 11-total power factor	9.84
MBUS 11-frequency	50.00

############################
not working

script:
>D 
>B 
=>sensor53 r
tper=10
>M 1 
+1,3,rE1,0,2400,MBUS,1,50,10400b4b16,107b0b8616,68030368530bB10f16
1,685d5d68080b72bcd8@1,11-Zähler-Nr.,,11_zähler,0
1,685d5d68080b72x14bcd8@100,11-total active energy,kWh,11_total_active_energy,2
1,685d5d68080b72x20bcd8@100,11-import active energy,kWh,11_import_active_energy,2
1,685d5d68080b72x26bcd8@100,11-export active energy,kWh,11_export_active_energy,2
1,685d5d68080b72x32bcd8@100,11-r.total active e,kWh,11_res_total_active_energy,2
1,685d5d68080b72x38bcd8@100,11-r.import active e,kWh,11_res_import_active_energy,2
1,685d5d68080b72x44bcd8@100,11-r.export active e,kWh,11_res_export_active_energy,2
1,685d5d68080b72x50bcd8@1,11-total reactive e,kWh,11_total_reactive_energy,1
1,685d5d68080b72x56bcd8@1,11-import reactive e,kWh,11_import_reactive_energy,1
1,685d5d68080b72x62bcd8@1,11-export reactive e,kWh,11_export_reactive_energy,1
1,685d5d68080b72x68bcd8@1,11-r. total reactive e,kWh,11_res_total_reactive_energy,1
1,685d5d68080b72x74bcd8@1,11-r. import reactive e,kWh,11_res_import_reactive_energy,1
1,685d5d68080b72x80bcd8@1,11-r. export reactive e,kWh,11_res_export_reactive_energy,1
1,68909068080b72x15bcd6@100,11-L1 voltage,V,11_L1_voltage,2
1,68909068080b72x21bcd6@100,11-L2 voltage,V,11_L2_voltage,2
1,68909068080b72x27bcd6@100,11-L3 voltage,V,11_L3_voltage,2
1,68909068080b72x33bcd6@100,11-L1L2 voltage,V,11_L1L2_voltage,2
1,68909068080b72x39bcd6@100,11-L2L3 voltage,V,11_L2L3_voltage,2
1,68909068080b72x45bcd6@100,11-L3L1 voltage,V,11_L3L1_voltage,2
1,68909068080b72x51bcd6@1000,11-L1 current,A,11_L1_current,2
1,68909068080b72x57bcd6@1000,11-L2 current,A,11_L2_current,2
1,68909068080b72x63cd6@1000,11-L3 current,A,11_L3_current,2
1,68909068080b72x69bcd6@1000,11-N current,A,11_N_current,2
1,68909068080b72x74bcd6@10,11-total active power,W,11_total_active_power,2
1,68909068080b72x79bcd6@10,11-L1 active power,W,11_L1_active_power,2
1,68909068080b72x84bcd6@10,11-L2 active power,W,11_L2_active_power,2
1,68909068080b72x89bcd6@10,11-L3 active power,W,11_L3_active_power,2
1,68909068080b72x119bcd4@100,11-total power factor,,11_total_power_factor,2
1,68909068080b72x139bcd4@100,11-frequency,,11_frequency,2

dump:
15:48:27.364 : 68 5d 5d 68 08 0b 72 65 79 24 01 24 40 01 02 55 00 00 00 0c 04 98 85 28 02 0c 04 41 80 28 02 0c 
15:48:27.510 : 04 57 05 00 00 0c 04 98 85 28 02 0c 04 41 80 28 02 0c 04 57 05 00 00 0c fd 3a 39 88 77 00 0c fd 
15:48:27.657 : 3a 42 48 13 00 0c fd 3a 96 39 64 00 0c fd 3a 39 88 77 00 0c fd 3a 42 48 13 00 0c fd 3a 96 39 64 
15:48:27.711 : 00 62 16 
15:48:33.018 : 68 90 90 68 08 0b 72 65 79 24 01 24 40 01 02 55 00 00 00 0b fd 47 61 36 02 0b fd 47 27 36 02 0b 
15:48:33.164 : fd 47 48 36 02 0b fd 47 53 09 04 0b fd 47 41 09 04 0b fd 47 72 09 04 0b fd 59 91 20 00 0b fd 59 
15:48:33.311 : 65 08 00 0b fd 59 40 21 00 0b fd 59 45 12 00 0b 2a 42 89 00 0b 2a 38 39 00 0b 2a 70 19 00 0b 2a 
15:48:33.457 : 56 30 00 0b fd 3a 29 16 f0 0b fd 3a 40 05 f0 0b fd 3a 29 02 00 0b fd 3a 22 13 f0 0a fd 3a 83 09 
15:48:33.598 : 0a fd 3a 90 09 0a fd 3a 93 09 0a fd 3a 17 09 0a fd 3a 04 50 72 16 

output:
MBUS 11-Zähler-Nr.	1247965
MBUS 11-total active energy	22885.97 kWh
MBUS 11-import active energy	22880.40 kWh
MBUS 11-export active energy	5.57 kWh
MBUS 11-r.total active e	22886 kWh
MBUS 11-r.import active e	1512880 kWh
MBUS 11-r.export active e	1211031 kWh
MBUS 11-total reactive e	20000001.0 kWh
MBUS 11-import reactive e	2000 kWh
MBUS 11-export reactive e	20466540 kWh
MBUS 11-r. total reactive e	128903248 kWh
MBUS 11-r. import reactive e	149013008 kWh
MBUS 11-r. export reactive e	74750009 kWh
MBUS 11-L1 voltage	236.88 V
MBUS 11-L2 voltage	236.62 V
MBUS 11-L3 voltage	236.59 V
MBUS 11-L1L2 voltage	410.06 V
MBUS 11-L2L3 voltage	15109.81 V
MBUS 11-L3L1 voltage	12110.31 V
MBUS 11-L1 current	100.00 A
MBUS 11-L2 current	0.01 A
MBUS 11-L3 current	0.00 A
MBUS 11-N current	1289.03 A
MBUS 11-total active power	9550.80 W
MBUS 11-L1 active power	103.10 W
MBUS 11-L2 active power	74657.40 W
MBUS 11-L3 active power	70795.40 W
MBUS 11-total power factor	61.74
MBUS 11-frequency	58.45

`

**Comment by gemu2015** [comment]
there is no known issue with sml in a file system setup.
however there are possible RAM problems on an esp8266 when defining a larger script buffer. 
you need to minimize RAM usage in your binary by removing all not needed stuff or use an esp32.

**Reply by killah78** [reply]
Thanks for your response. Yes I sadly found out that it makes no sense with a 8266 and a larger script. It's not usable anymore. I switched to an esp32c3 which is currently running fine. But still testing. Thanks.
